openvdb_houdini/SOP_NodeVDB.cc

```diff
@@ -711,13 +711,48 @@ OpenVDBOpFactory::OpenVDBOpFactory(
     setNativeName(OpenVDBOpPolicy().getNativeName(*this));
 }
 
+OpenVDBOpFactory::~OpenVDBOpFactory()
+{
+    if (!mNativeName.empty()) {
+
+        // environment variable takes precedence over compiler flag
+
+        const char* opHidePolicy = std::getenv("OPENVDB_OPHIDE_POLICY");
+
+#ifdef OPENVDB_OPHIDE_POLICY
+        if (opHidePolicy == nullptr) {
+            opHidePolicy = OPENVDB_PREPROC_STRINGIFY(OPENVDB_OPHIDE_POLICY);
+        }
+#endif
+
+        if (opHidePolicy != nullptr) {
+            std::string opHidePolicyStr(opHidePolicy);
+
+            // to lower-case
+
+            std::transform(opHidePolicyStr.begin(), opHidePolicyStr.end(),
+                opHidePolicyStr.begin(),
+                [](unsigned char c) { return std::tolower(c); });
+
+            if (opHidePolicyStr == "aswf") {
+                // set this SOP to be hidden (if a native equivalent exists)
+                this->setInvisible();
+            } else if (opHidePolicyStr == "native") {
+                // mark the native equivalent SOP to be hidden
+                this->addInvisibleName(mNativeName);
+            }
+        }
+
+        addSpareData({{"nativename", mNativeName}});
+    }
+}
 
 OpenVDBOpFactory&
 OpenVDBOpFactory::setNativeName(const std::string& name)
 {
     // SideFX nodes have no native equivalent.
 #ifndef SESI_OPENVDB
-    addSpareData({{"nativename", name}});
+    mNativeName = name;
```

Comment:
@pcucka - It's possible I missed something, but I didn't see a way to retrieve the "nativename" value from the base class, so I've modified this implementation slightly to re-introduce the native name member variable and then add the spare data in the deconstructor of this class.

Reply:
All of the logic that's currently in the destructor can be moved into `setNativeName`, can't it?

Reply:
No because setNativeName() is called multiple times for many operators - once from the OpenVDBOpFactory constructor and then overridden from the operator registration, which would add multiple hide names to the OP_Table. The only option to handle this cleanly would be to remove the old name from the OP_Table when it is modified which seems like a lot of extra redundant logic.

Reply:
I don't see any problem with checking whether the native SOP exists before changing visibilities or with calling `OP_OperatorTable::delOpHidden` to undo a change.

Reply:
Ok, then I shall do it like that.